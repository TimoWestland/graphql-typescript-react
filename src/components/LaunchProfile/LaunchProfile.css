.launch-profile {
  height: 100vh;
  max-height: 100%;
  width: calc(100vw - 300px);
  overflow: hidden auto;
  padding-left: 20px;
  padding-right: 20px;
}

.launch-profile__status {
  margin-top: 40px;
  margin-bottom: 30px;
}

.launch-profile__title {
  margin-top: 0;
  margin-bottom: 4px;
}

.launch-profile__success {
  color: #2cb84b;
}

.launch-profile__failed {
  color: #ff695e;
}

.launch-profile__image-list {
  display: grid;
  grid-gap: 20px;
  grid-template-columns: repeat(2, 1fr);
  margin-top: 40px;
  padding-bottom: 100px;
}

.launch-profile__image {
  width: 100%;
}
