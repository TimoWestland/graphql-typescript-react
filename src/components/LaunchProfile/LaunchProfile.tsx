import './LaunchProfile.css'

import * as React from 'react'
import { LaunchProfileQuery } from '../../generated/graphql'

export interface Props {
  data: LaunchProfileQuery
}

export const LaunchProfile: React.FC<Props> = ({ data }) => {
  if (!data.launch) {
    return <div>No launch available</div>
  }

  const { launch } = data

  return (
    <div className="launch-profile">
      <div className="launch-profile__status">
        <span>Flight {launch.flight_number}: </span>
        {launch.launch_success ? (
          <span className="launch-profile__success">Success</span>
        ) : (
          <span className="launch-profile__failed">Failed</span>
        )}
      </div>
      <h1 className="launch-profile__title">
        {launch.mission_name}
        {launch.rocket && ` (${launch.rocket.rocket_name} | ${launch.rocket.rocket_type})`}
      </h1>
      <p className="launch-profile__description">{launch.details}</p>
      {!!launch.links && !!launch.links.flickr_images && (
        <div className="launch-profile__image-list">
          {launch.links.flickr_images.map((image) =>
            image ? <img src={image} className="launch-profile__image" key={image}  alt="yolo"/> : null
          )}
        </div>
      )}
    </div>
  )
}

export default LaunchProfile
