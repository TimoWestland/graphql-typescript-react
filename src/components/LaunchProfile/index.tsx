import * as React from 'react'
import { useLaunchProfileQuery } from '../../generated/graphql'
import LaunchProfile from './LaunchProfile'

interface Props {
  id: number
}

const LaunchProfileContainer: React.FC<Props> = ({ id }) => {
  const { data, error, loading, refetch } = useLaunchProfileQuery({
    variables: { id: String(id) },
  })

  React.useEffect(() => {
    refetch()
  }, [id, refetch])

  if (loading) {
    return <div>Loading...</div>
  }

  if (error) {
    return <div>Something went wrong!</div>
  }

  if (!data) {
    return <div>Select a flight from the panel on the left.</div>
  }

  return <LaunchProfile data={data} />
}

export default LaunchProfileContainer
