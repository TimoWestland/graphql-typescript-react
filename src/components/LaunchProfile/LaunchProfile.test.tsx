import * as React from 'react'
import { render, cleanup } from '@testing-library/react'
import { LaunchProfileQuery } from '../../generated/graphql'
import LaunchProfile, { Props } from './LaunchProfile'

const mockLaunch: LaunchProfileQuery['launch'] = {
  flight_number: 1,
  mission_name: 'Some name',
  launch_year: 2019,
  launch_success: true,
  details: 'Some more information about this launch.',
  launch_site: {
    site_name: 'Almere haven',
  },
  rocket: {
    rocket_name: 'Rocket man',
    rocket_type: 'Elton John',
  },
  links: {
    flickr_images: ['/some/src1.png', '/some/src2.png'],
  },
}

describe('LaunchProfile', () => {
  const defaultProps: Props = {
    data: {
      launch: mockLaunch,
    },
  }

  const create = (props: Partial<Props> = {}) =>
    render(<LaunchProfile {...defaultProps} {...props} />)

  afterEach(() => {
    cleanup()
  })

  it('should match snapshot', () => {
    const { container } = create()
    expect(container).toMatchSnapshot()
  })

  it('should render failed status if launch failed', () => {
    const { getByText } = create({
      data: {
        launch: {
          ...mockLaunch,
          launch_success: false
        },
      },
    })
    const statusFailed = getByText('Failed')
    expect(statusFailed).toBeInTheDocument()
  })
})
