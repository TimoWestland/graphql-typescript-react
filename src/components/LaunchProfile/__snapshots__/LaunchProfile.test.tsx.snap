// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchProfile should match snapshot 1`] = `
<div>
  <div
    class="launch-profile"
  >
    <div
      class="launch-profile__status"
    >
      <span>
        Flight 
        1
        : 
      </span>
      <span
        class="launch-profile__success"
      >
        Success
      </span>
    </div>
    <h1
      class="launch-profile__title"
    >
      Some name
       (Rocket man | Elton John)
    </h1>
    <p
      class="launch-profile__description"
    >
      Some more information about this launch.
    </p>
    <div
      class="launch-profile__image-list"
    >
      <img
        alt="yolo"
        class="launch-profile__image"
        src="/some/src1.png"
      />
      <img
        alt="yolo"
        class="launch-profile__image"
        src="/some/src2.png"
      />
    </div>
  </div>
</div>
`;
